contracts/vortex/CarbonVortex.sol

```diff
@@ -3,58 +3,102 @@ pragma solidity 0.8.19;
 
 import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
 import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
 
 import { ICarbonVortex } from "./interfaces/ICarbonVortex.sol";
 import { IVersioned } from "../utility/interfaces/IVersioned.sol";
+import { IVault } from "../utility/interfaces/IVault.sol";
 import { ICarbonController } from "../carbon/interfaces/ICarbonController.sol";
 import { Upgradeable } from "../utility/Upgradeable.sol";
-import { Token } from "../token/Token.sol";
+import { Token, NATIVE_TOKEN } from "../token/Token.sol";
 import { Utils } from "../utility/Utils.sol";
 import { MathEx } from "../utility/MathEx.sol";
-import { MAX_GAP, PPM_RESOLUTION } from "../utility/Constants.sol";
-
-interface IBancorNetwork {
-    function collectionByPool(Token pool) external view returns (address);
-
-    function tradeBySourceAmount(
-        Token sourceToken,
-        Token targetToken,
-        uint256 sourceAmount,
-        uint256 minReturnAmount,
-        uint256 deadline,
-        address beneficiary
-    ) external payable returns (uint256);
-}
+import { ExpDecayMath } from "../utility/ExpDecayMath.sol";
+import { PPM_RESOLUTION, MAX_GAP } from "../utility/Constants.sol";
 
 /**
- * @dev CarbonVortex contract
+ * @notice CarbonVortex contract
  */
 contract CarbonVortex is ICarbonVortex, Upgradeable, ReentrancyGuardUpgradeable, Utils {
     using Address for address payable;
+    using SafeCast for uint256;
+
+    uint128 private constant INITIAL_PRICE_SOURCE_AMOUNT = type(uint128).max;
+    uint128 private constant INITIAL_PRICE_TARGET_AMOUNT = 1e12;
```

Comment:
What's the reasoning behind this value?

Reply:
Goal here is to start from the maximum possible auction price to encompass all of the different token prices. Auction starts at max price and progresses slowly via exp decay based on the specified hafllife (currently set to 12 hours).
After a while for each token, market price should be reached and the tokens should be open for arbing if market conditions allow it (profit from the arb exceed the gas costs)